lib/OAuth2.php

```diff
@@ -1001,8 +997,7 @@ protected function grantAccessTokenExtension(IOAuth2Client $client, array $input
         if (!($this->storage instanceof IOAuth2GrantExtension)) {
             throw new OAuth2ServerException(self::HTTP_BAD_REQUEST, self::ERROR_UNSUPPORTED_GRANT_TYPE);
         }
-        $uri = filter_var($inputData["grant_type"], FILTER_VALIDATE_URL);
```

Comment:
You may check both URI and URN.

A regex pattern for URN could be something like `^urn:[a-z0-9][a-z0-9-]{0,31}:[a-z0-9()+,\-.:=@;$_!*'%/?#]+$`